### ISSUE TITLE
Memory requirement?

### ISSUE THREAD

**Issue by Rajmehta123** [Yuliang-Liu/Monkey#34]
@echo840  I am using the demo.py but the model goes OOM even after having 96Gb of GPU memory. Looks like it is only using single gpu and not distributing the model into multiple GPUs.

**Comment by ShuoZhang2003** [comment]
The code provided uses single GPU inference by default. You can try to modify the code to deploy the model on multiple GPUs.

**Comment by echo840** [comment]
It only needs around 30g gpu memory. You can use this code for inference.
'''Python
from transformers import AutoModelForCausalLM, AutoTokenizer
checkpoint = "echo840/Monkey"
model = AutoModelForCausalLM.from_pretrained(checkpoint, device_map='cuda', trust_remote_code=True).eval()
tokenizer = AutoTokenizer.from_pretrained(checkpoint, trust_remote_code=True)
tokenizer.padding_side = 'left'
tokenizer.pad_token_id = tokenizer.eod_id
img_path = ""
question = ""
query = f'<img>{img_path}</img> {question} Answer: ' #VQA
# query = f'<img>{img_path}</img> Generate the detailed caption in English: ' #detailed caption

input_ids = tokenizer(query, return_tensors='pt', padding='longest')
attention_mask = input_ids.attention_mask
input_ids = input_ids.input_ids

pred = model.generate(
            input_ids=input_ids.cuda(),
            attention_mask=attention_mask.cuda(),
            do_sample=False,
            num_beams=1,
            max_new_tokens=512,
            min_new_tokens=1,
            length_penalty=1,
            num_return_sequences=1,
            output_hidden_states=True,
            use_cache=True,
            pad_token_id=tokenizer.eod_id,
            eos_token_id=tokenizer.eod_id,
            )
response = tokenizer.decode(pred[0][input_ids.size(1):].cpu(), skip_special_tokens=True).strip()
print(response)
'''
